rkt/image/fetcher.go

```diff
@@ -243,51 +281,42 @@ func (f *Fetcher) fetchSingleImageByPath(path string, a *asc) (string, error) {
 	return ff.Hash(path, a)
 }
 
-type appBundle struct {
-	App *discovery.App
-	Str string
-}
-
-func newAppBundle(name string) (*appBundle, error) {
-	app, err := discovery.NewAppFromString(name)
-	if err != nil {
-		return nil, errwrap.Wrap(fmt.Errorf("invalid image name %q", name), err)
-	}
+// TODO(sgotti) I'm not sure setting default os and arch also for image
+// dependencies is correct since it may break noarch dependencies. Reference:
+// https://github.com/coreos/rkt/pull/2317
+func (db *distBundle) setAppDefaults() error {
+	app := db.dist.(*dist.Appc).App()
 	if _, ok := app.Labels["arch"]; !ok {
 		app.Labels["arch"] = runtime.GOARCH
```

Comment:
Note to self: this will need to be adjusted following https://github.com/appc/spec/pull/668.